aten/src/ATen/autocast_mode.cpp

```diff
@@ -389,6 +389,8 @@ TORCH_LIBRARY_IMPL(aten, Autocast, m) {
   KERNEL(gru_cell, lower_precision_fp)
   KERNEL(rnn_tanh_cell, lower_precision_fp)
   KERNEL(rnn_relu_cell, lower_precision_fp)
+  KERNEL(_scaled_dot_product_flash_attention, lower_precision_fp)
+  KERNEL(_scaled_dot_product_attention, lower_precision_fp)
```

Comment:
I think we need only `_scaled_dot_product_flash_attention`, also we need a dedicated test (because previously this failure wasn't caught).

Reply:
I updated the PR to include tests that test that flash is failing when run with FP32 and works with autocast to float16 and bfloat16